src/main/java/org/gridsuite/dynamicsecurityanalysis/server/controller/DynamicSecurityAnalysisController.java

```diff
@@ -0,0 +1,141 @@
+
+        DynamicSecurityAnalysisRunContext dynamicSecurityAnalysisRunContext = parametersService.createRunContext(
+            networkUuid,
+            variantId,
+            receiver,
+            provider,
+            ReportInfos.builder().reportUuid(reportId).reporterId(reportName).computationType(reportType).build(),
+            userId,
+            dynamicSimulationResultUuid,
+            parametersUuid);
+
+        UUID resultUuid = dynamicSecurityAnalysisService.runAndSaveResult(dynamicSecurityAnalysisRunContext);
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(resultUuid);
+    }
+
+    @GetMapping(value = "/results/{resultUuid}/status", produces = "application/json")
+    @Operation(summary = "Get the dynamic security analysis status from the database")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The dynamic simulation status"),
+        @ApiResponse(responseCode = "204", description = "Dynamic security analysis status is empty"),
+        @ApiResponse(responseCode = "404", description = "Dynamic security analysis result uuid has not been found")})
+    public ResponseEntity<DynamicSecurityAnalysisStatus> getStatus(@Parameter(description = "Result UUID") @PathVariable("resultUuid") UUID resultUuid) {
+        DynamicSecurityAnalysisStatus result = dynamicSecurityAnalysisService.getStatus(resultUuid);
+        return ResponseEntity.ok().body(result);
+    }
+
+    @PutMapping(value = "/results/invalidate-status", produces = "application/json")
+    @Operation(summary = "Invalidate the dynamic security analysis status from the database")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The dynamic security analysis result uuids have been invalidated"),
+        @ApiResponse(responseCode = "404", description = "Dynamic security analysis result has not been found")})
+    public ResponseEntity<List<UUID>> invalidateStatus(@Parameter(description = "Result UUIDs") @RequestParam("resultUuid") List<UUID> resultUuids) {
+        List<UUID> result = dynamicSecurityAnalysisResultService.updateStatus(resultUuids, DynamicSecurityAnalysisStatus.NOT_DONE);
+        return CollectionUtils.isEmpty(result) ? ResponseEntity.notFound().build() :
+                ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(result);
+    }
+
+    @DeleteMapping(value = "/results/{resultUuid}")
+    @Operation(summary = "Delete a dynamic security analysis result from the database")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The dynamic simulation result has been deleted")})
+    public ResponseEntity<Void> deleteResult(@Parameter(description = "Result UUID") @PathVariable("resultUuid") UUID resultUuid) {
+        dynamicSecurityAnalysisResultService.delete(resultUuid);
+        return ResponseEntity.ok().build();
+    }
+
+    @DeleteMapping(value = "/results", produces = APPLICATION_JSON_VALUE)
```

Comment:
We have this endpoint for other computation servers. But it's dangerous and we don't use it I think

Reply:
whether we have some admin tool call to this endpoint? I see in LoadFlow server, we have an endpoint which accept a list of ids but also null list to delete all..

Reply:
Ok yes, it's called by some admin tools scripts